.github/workflows/main.yml

```diff
@@ -0,0 +1,348 @@
+name: CI
+
+on:
+  # TODO: add nightly schedule.
+  pull_request:
+    branches:
+    - master
+  push:
+    branches:
+    - master
+  release:
+    types: [published]
+
+jobs:
+  check_core:
+    name: Check Core
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python: ['3.6', '3.7']
+
+    steps:
+    - uses: actions/checkout@v2
+
+    - name: Setup Python
+      uses: actions/setup-python@v1
+      with:
+        python-version: ${{ matrix.python }}
+
+    - uses: actions/cache@v1
+      with:
+        path: ${{ env.pythonLocation }}
+        key: ${{ runner.os }}-pydeps-${{ matrix.python }}-${{ hashFiles('setup.py') }}-${{ hashFiles('dev-requirements.txt') }}
+        restore-keys: |
+          ${{ runner.os }}-pydeps-${{ matrix.python }}
+
+    - name: Install requirements
+      run: |
+        pip install --upgrade pip setuptools wheel
+        pip install -e .
+        pip install --upgrade -r dev-requirements.txt
+
+    - name: Debug info
+      run: |
+        pip freeze
+
+    - name: Lint
+      run: |
+        make lint
+
+    - name: Type check
+      run: |
+        make typecheck
+
+    - name: Run tests
```

Comment:
What about also adding a `python setup.py check`? This performs checks package-wise, especially on the metadata.

Reply:
That's probably a good idea. It looks like it will fail though when we have a `-unreleased` version?

Reply:
Yeah, it breaks pip and PyPI. There are some rules of what are valid suffixes.